lib/frameworks/capybara.rb

```diff
@@ -226,36 +225,51 @@ def register_mechanize_driver(opts)
     :mechanize
   end
 
-
   def register_poltergeist_driver(opts)
-    ## Poltergiest needs a Rack application: create a dummy one
-    app = Proc.new do |env|
-      ['200', {'Content-Type' => 'text/html'}, ['A barebones rack app.']]
+    # Poltergiest needs a Rack application: create a dummy one
+    app = Proc.new do
+      ['200', { 'Content-Type' => 'text/html' }, ['A barebones rack app.']]
     end
     phantom_opts = %w(--ignore-ssl-errors=true)
     phantom_opts.push "--ssl-client-certificate-file=#{ENV['FW_CERT_LOCATION']}" if ENV['FW_CERT_LOCATION']
     phantom_opts.push "--proxy=#{@proxy_host}:#{@proxy_port}" if @proxy_host && @proxy_port
     Capybara.app = app
     Capybara.run_server = false
     options = {
-      js_errors: false,
+      js_errors: true,
       timeout:   120,
       window_size: [1200, 1000],
       phantomjs_options: phantom_opts,
       default_wait_time: 30
     }
     options[:phantomjs] = ENV['PHANTOMJS_PATH'] if ENV['PHANTOMJS_PATH'] && !ENV['PHANTOMJS_PATH'].empty?
     Capybara.register_driver :poltergeist do |app|
-      Capybara.app_host = "http://www.bbc.co.uk"
+      Capybara.app_host = 'http://www.bbc.co.uk'
       Capybara::Poltergeist::Driver.new(app, options)
     end
     :poltergeist
   end
 
+  def register_headless_chromium_driver(_opts)
+    app = Proc.new do
+      ['200', { 'Content-Type' => 'text/html' }, ['A barebones rack app.']]
+    end
+    Capybara.register_driver :headless_chromium do |app|
+      Capybara.app_host = 'http://www.bbc.co.uk'
+      caps = Selenium::WebDriver::Remote::Capabilities.chrome(
+        'chromeOptions' => {
+          'binary' => '/Users/tabibs01/Downloads/chrome-linux/chrome',
```

Comment:
Need to find the solution to make it generic location for e.g. keeping it in repo, thoughts ?